### DISCUSSION TITLE
How to access the GUI

### DISCUSSION THREAD

**Discussion by BenJamesAndo** [wanghaisheng/tiktoka-studio-uploader#14] [Q&A] [answered]
Hello. I see you're working on a GUI. I tried it without a GUI and couldn't really figure it out. The GUI looks like it'd make it much easier for some people. To get access you say you need to become a sponsor. Is that for a sponsorship of any amount, or does it have to be a certain amount?

Also is the GUI translated into English? In the video it wasn't completely in English.

**Comment by wanghaisheng** [comment, marked as answer]
yes  any tier should be ok during these days since you are the initial ones
and it is a complete english version .as following i record yesterday
https://youtu.be/Xh-Dmm1POBo

**Reply by BenJamesAndo** [reply]
Ok great. I've signed up to be your first sponsor here on Github. Please send the GUI when you can.
Thanks.

**Reply by wanghaisheng** [reply]
ok  
perhaps I will setup a discord lately  where bot can serve all you guys

**Reply by wanghaisheng** [reply]
@BenJamesAndo  I could not find your email pls dm me
tiktokadownloader@gmail.com

**Reply by BenJamesAndo** [reply]
I've emailed you now.

**Reply by wanghaisheng** [reply]
any issue you encounter 
I need your TESTIMONIALS

**Reply by noone-noone** [reply]
Hello would like ask whether the access to gui is still available guess

**Reply by wanghaisheng** [reply]
@noone-noone  I have tried to build a new modern version but still not release yet.
I can provide you the old to test

**Reply by noone-noone** [reply]
Hello,
Thanks 
Does old version work fine?  If yes, Please provide it

**Reply by wanghaisheng** [reply]
I have not tested it lately, but I promise tomorrow I will test it and mail to you as soon as possible .@noone-noone

**Reply by noone-noone** [reply]
Hello thanks sure will patiently wait
In fact, I guess I could test it myself.  But either way, Will wait

**Comment by noone-noone** [comment]
Have tried send you message from my email tiktokadownloader@gmail.com, But gmail blocked it for some reason
So sent you an email from my gmail address

**Reply by wanghaisheng** [reply]
I just change the old [tiktokadownloader@gmail.com](mailto:tiktokadownloader@gmail.com) to my custom email 
admin@tiktokastudio.com, can you try this one

**Reply by noone-noone** [reply]
Hello, Thanks
have sent message now
Hope you received it
will wait

**Comment by noone-noone** [comment]
Hello
Would like ask whether you have any updates on this?

**Reply by wanghaisheng** [reply]
as you can see, I just fixed the underlying lib for the GUI
I will look at the GUI code and integrate them together @noone-noone

**Reply by noone-noone** [reply]
Hello, 
thanks sure will patiently wait 
But there is something I dont quite understand
The uploaded script is for youtube and Douyin 
But will it also work with tiktok?

**Reply by wanghaisheng** [reply]
the core  lib I have developed since last year, the old gui only integrate youtube. tiktok lib is still under developed
today I have tested and upgraded some code of core lib

**Reply by noone-noone** [reply]
So currently tiktok not supported at all? When will you be able to add? 
And what about the command line version? It does not support tiktok as well?

**Reply by wanghaisheng** [reply]
when I provide the old gui to you ,i will look at the core tiktok lib and try to add it to gui version ,that is my plan

**Reply by noone-noone** [reply]
Hello
Currently tiktok lib already work? or not yet?

**Reply by noone-noone** [reply]
Just would like ask whether currently the tiktok lib can be used in any way? like onefile-example-douyin for tiktok does not currently work?

**Reply by wanghaisheng** [reply]
I  have tested douyin  and I  think there need testing and fix work  based on douyin scripts to be done.I will try to write a onefile for tiktok

**Reply by noone-noone** [reply]
I understand guess that apparently Douyin and Tiktok are similar
Douyin is Tiktok but are they similar in structure? I mean, Simply replacing the URL from duoyin to tiktok would be enough? or it is harder than that?

**Reply by wanghaisheng** [reply]
I have  not checked that, tiktok have a annoying capacha  we are facing ,other things wont be much work

**Reply by noone-noone** [reply]
Hello thanks have seen certain tiktok captcha solvers on Github, But not checked whether they are working or not 
https://github.com/lumaaaaaa/ttCaptchaSolver/tree/master
https://github.com/xtekky/TikTok-Captcha-Solver
etc
I have not tested whether any of them are working or not, So really unsure whether it will help

**Reply by wanghaisheng** [reply]
I have written one for douyin and I will test it against tiktok

**Reply by noone-noone** [reply]
Hello thanks hope the script will work

**Reply by wanghaisheng** [reply]
today I dig a little about tiktok upload .tomorrow will write a first draft

**Reply by noone-noone** [reply]
Hello thanks does it works?

**Reply by wanghaisheng** [reply]
it kinda of crack thing ,using a reverse api , works but I have concerns about account banning etc

**Reply by noone-noone** [reply]
What exactly does the script require? api key?

**Reply by wanghaisheng** [reply]
nope. I will make an example tomorrow~

**Reply by noone-noone** [reply]
Hello thanks hope will figure it out

**Reply by wanghaisheng** [reply]
@noone-noone  good news
I have managed to fix the last issue of youtube upload (after schedule time, the done button wont click).
for tiktok part, managed to use stealth to avoid capcha appear.and it works.
I need to use stealth to auto get cookie file saved  when you first manually login()
and there is a selenium lib  i can learn 
https://github.com/wkaisertexas/tiktok-uploader

**Reply by noone-noone** [reply]
Hello 
Thanks this look great but currently trying to follow your instruction and apparently the firstrun files are missing
Could you please clarify this?

**Reply by wanghaisheng** [reply]
i  am dramatically rebuild this. I would change it now
it is savecookie now

**Reply by noone-noone** [reply]
> i am dramatically rebuild this. I would change it now it is savecookie now

Hello thanks guess now could not find any information regarding the tiktok upload in the readme.
I am probably doing something wrong, But when trying to run the onefile-example-youtube I am getting the following error:
"start to check Tiktoka Studio requirements whether  intalled
start to check Tiktoka Studio requirements whether playwright intalled
start to check Tiktoka Studio requirements whether playwright browser intalled
import pl library
initial pl library
start to check Tiktoka Studio requirements whether chromium intalled
start to check Tiktoka Studio requirements whether webkit/edge intalled
start to check Tiktoka Studio requirements whether firefox intalled
Tiktoka Studio requirements-playwright have intalled
Tiktoka Studio requirements-browser have intalled
default closewhen100percent:0
publishpolicy:2
CaptionsCertification:0
LicenceType:0
ShortsremixingType:0
CommentsRatingsPolicy:1
DEBUG: whether run in view mode:True
using stealth
stealth is latest
b:\tt\ytb_up\youtube.py:220: RuntimeWarning: coroutine 'PlaywrightAsyncDriver.__exit__' was never awaited
  with PlaywrightAsyncDriver(
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
DEBUG: firefox is now running with proxy:socks5://127.0.0.1:1080
第 1 次初始化浏览器, 端口为:7329
初始化浏览器异常: NS_ERROR_PROXY_CONNECTION_REFUSED
=========================== logs ===========================
navigating to "https://bot.sannysoft.com/", waiting until "load"
============================================================
Traceback (most recent call last):
  File "b:\tt\ytb_up\login.py", line 207, in botcheck
    await self.page.goto(url)
  File "C:\Users\mikhaelb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\async_api\_generated.py", line 9204, in goto
    await self._impl_obj.goto(
  File "C:\Users\mikhaelb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\_impl\_page.py", line 495, in goto
    return await self._main_frame.goto(**locals_to_params(locals()))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mikhaelb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\_impl\_frame.py", line 147, in goto
    await self._channel.send("goto", locals_to_params(locals()))
  File "C:\Users\mikhaelb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\_impl\_connection.py", line 61, in send
    return await self._connection.wrap_api_call(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mikhaelb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\_impl\_connection.py", line 461, in wrap_api_call
    return await cb()
           ^^^^^^^^^^
  File "C:\Users\mikhaelb\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\playwright\_impl\_connection.py", line 96, in inner_send
    result = next(iter(done)).result()
             ^^^^^^^^^^^^^^^^^^^^^^^^^
playwright._impl._api_types.Error: NS_ERROR_PROXY_CONNECTION_REFUSED
=========================== logs ===========================
navigating to "https://bot.sannysoft.com/", waiting until "load"
============================================================

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "b:\tt\onefile-example_youtube.py", line 266, in <module>
    scheduletopublish_tomorrow()
  File "b:\tt\onefile-example_youtube.py", line 184, in scheduletopublish_tomorrow
    asyncio.run(
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1008.0_x64__qbz5n2kfra8p0\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1008.0_x64__qbz5n2kfra8p0\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1008.0_x64__qbz5n2kfra8p0\Lib\asyncio\base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "b:\tt\ytb_up\youtube.py", line 238, in upload
    await botcheck(pl)
  File "b:\tt\ytb_up\login.py", line 220, in botcheck
    self.browser.quit()
    ^^^^^^^^^^^^^^^^^
AttributeError: 'Browser' object has no attribute 'quit'
Exception ignored in: <function BaseSubprocessTransport.__del__ at 0x000002395FA459E0>
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1008.0_x64__qbz5n2kfra8p0\Lib\asyncio\base_subprocess.py", line 126, in __del__
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1008.0_x64__qbz5n2kfra8p0\Lib\asyncio\base_subprocess.py", line 104, in close
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1008.0_x64__qbz5n2kfra8p0\Lib\asyncio\proactor_events.py", line 109, in close
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1008.0_x64__qbz5n2kfra8p0\Lib\asyncio\base_events.py", line 761, in call_soon
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1008.0_x64__qbz5n2kfra8p0\Lib\asyncio\base_events.py", line 519, in _check_closed
RuntimeError: Event loop is closed"

**Reply by noone-noone** [reply]
Well probably have not configured it properly
also will it run on windows or it should be run from linux guess?

**Reply by wanghaisheng** [reply]
wait a little,  i work on windows and mac mini
